### PR TITLE
feat(tree): enable custom filter of query

### PR DESCRIPTION
**Pull request by wattachai-lseg**
## Description

Combo Box, Tree & Tree Select support filtering items with their latest prop values. For Tree & Tree Select, EF docs show how to do so with Tree Node. Note that Combo Box is an exception. As Combo Box has no API editing prop values of items directly, custom filter for Combo Box is limited to initial prop values.

* Expose Tree's `filter` property for custom filter of query
* Add missing API reference for custom filter to Tree Select docs
* Improve Combo Box docs on custom filter

https://jira.refinitiv.com/browse/DME-2141

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by Theeraphat-Sorasetsakul — DISMISSED** [review]
LGTM.

File: packages/elements/src/combo-box/__demo__/index.html [hunk 1]
```diff
@@ -170,10 +170,10 @@
 
           // return scoped custom filter
           return (item) => {
+            const value = item.value;
+            const label = item.label;
             const regex = getRegularExpressionOfQuery();
-            // test on label or value
-            const result = query === item.value || regex.test(item.label);
-            regex.lastIndex = 0; // do not forget to reset last index
+            const result = regex.test(value) || regex.test(label);
```
Comment (bualoy-napat) [inline packages/elements/src/combo-box/__demo__/index.html:176]: I am just curious why demo in v7 doesn't require to reset lastIndex but v6 does

File: packages/elements/src/combo-box/__test__/combo-box.filter.test.js [hunk 2]
```diff
@@ -40,5 +42,37 @@ describe('combo-box/Filter', function () {
       expect(el.query).to.equal(textInput, 'Query should be the same as input text: "Aland Islands"');
       await expect(el).shadowDom.to.equalSnapshot(snapshotIgnore);
     });
+
+    it('Should be able to use custom filter function', async function () {
```
Comment (bualoy-napat) [inline packages/elements/src/combo-box/__test__/combo-box.filter.test.js:46]: This test case doesn't available v6. Please port this test case to v6.

File: packages/elements/src/combo-box/__demo__/index.html (hunk 1 above)
Comment (wattachai-lseg) [reply packages/elements/src/combo-box/__demo__/index.html:176]: Nice catch! They are all removed. Note that, it applies to both versions.

File: packages/elements/src/combo-box/__test__/combo-box.filter.test.js (hunk 2 above)
Comment (wattachai-lseg) [reply packages/elements/src/combo-box/__test__/combo-box.filter.test.js:46]: Another good catch! Added.

**Review by bualoy-napat — APPROVED** [review]
Verified

**Pull request merged**